dist/scnnr.cjs.js

```diff
@@ -198,11 +198,140 @@ var errors = Object.freeze({
 	RecognitionError: RecognitionError
 });
 
+var AuthInterceptor = function () {
+  function AuthInterceptor() {
+    classCallCheck(this, AuthInterceptor);
+    this.interceptRequest = this.interceptRequest.bind(this);
+  }
+
+  createClass(AuthInterceptor, [{
+    key: "interceptRequest",
+    value: function interceptRequest(config) {
+      return Promise.resolve(config);
+    }
+  }]);
+  return AuthInterceptor;
+}();
+
+var PrivateKeyAuthInterceptor = function (_AuthInterceptor) {
+  inherits(PrivateKeyAuthInterceptor, _AuthInterceptor);
+
+  function PrivateKeyAuthInterceptor(apiKey) {
+    classCallCheck(this, PrivateKeyAuthInterceptor);
+
+    var _this = possibleConstructorReturn(this, (PrivateKeyAuthInterceptor.__proto__ || Object.getPrototypeOf(PrivateKeyAuthInterceptor)).call(this));
+
+    _this.apiKey = apiKey;
+    return _this;
+  }
+
+  createClass(PrivateKeyAuthInterceptor, [{
+    key: 'interceptRequest',
+    value: function interceptRequest(config) {
+      var _this2 = this;
+
+      return new Promise(function (resolve, reject) {
+        config.headers['x-api-key'] = _this2.apiKey;
+        resolve(config);
+      });
+    }
+  }]);
+  return PrivateKeyAuthInterceptor;
+}(AuthInterceptor);
+
+var PublicKeyAuthInterceptor = function (_AuthInterceptor) {
+  inherits(PublicKeyAuthInterceptor, _AuthInterceptor);
+
+  function PublicKeyAuthInterceptor(publicAPIKey, options) {
+    classCallCheck(this, PublicKeyAuthInterceptor);
+
+    var _this = possibleConstructorReturn(this, (PublicKeyAuthInterceptor.__proto__ || Object.getPrototypeOf(PublicKeyAuthInterceptor)).call(this));
+
+    _this.publicAPIKey = publicAPIKey;
+    _this.options = options;
+    _this.marginToExpire = 0.05; // a margin to prevent unexpected expiration (5% of the time)
+    return _this;
+  }
+
+  createClass(PublicKeyAuthInterceptor, [{
+    key: 'interceptRequest',
+    value: function interceptRequest(config) {
+      return this.getOneTimeToken().then(function (token) {
+        config.headers['x-api-key'] = 'use-scnnr-one-time-token';
+        config.headers['x-scnnr-one-time-token'] = token;
```

Comment:
Your implementation seems to work well! The following comment is maybe for a future PR on the scnnr project.

I think maybe our documentation needs to be updated here: https://api.scnnr.cubki.jp/v1/docs#tag/authorization%2Fpaths%2F~1auth~1tokens%2Fpost

Even though the header field is `x-scnnr-one-time-token` (which is used successfully in this implementation),
the documentation specifies it as `use-x-scnnr-one-time-token` and as `-H 'use-scnnr-one-time-token: ..`

Reply:
Exactly. That is the problem of our API server project.
And I already fixed it, but we don't deploy it yet.
I'll plan to deploy it as soon as possible, so please be patient. 😅